src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -262,6 +263,18 @@ private PodTemplateCallback(PodTemplate podTemplate) {
         protected void finished(StepContext context) throws Exception {
             try {
                 KubernetesCloud cloud = resolveCloud();
+                LOGGER.log(Level.FINE, () -> "Terminating nodes of pod template " + podTemplate.getName());
+                Jenkins.get().getNodes()
+                        .stream()
+                        .filter(KubernetesSlave.class::isInstance)
+                        .map(KubernetesSlave.class::cast)
+                        .forEach(node -> {
+                            try {
+                                node.terminate();
+                            } catch (InterruptedException | IOException e) {
+                                LOGGER.log(Level.WARNING, "Failed to terminate " + node.getNodeName(), e);
+                            }
+                        });
```

Comment:
This is removing ALL KubernetesSlave nodes, no? In my Jenkins, I have many jobs running at once, and each job declares it's own podTemplate step. How is this not terminating all such nodes?

Reply:
Indeed, this would need to filter based on pod template, and if (after resolving merge conflicts) this passes all existing tests then a new test would need to be written which runs two `podTemplate` steps in parallel and lets one finish and asserts that the other continues running.